clash-lib/src/Data/Aeson/Extra.hs

```diff
@@ -25,7 +25,12 @@ import           Data.List            (intercalate)
 import           Data.List.NonEmpty   (NonEmpty (..))
 import qualified Data.List.NonEmpty   as NE
 import           Data.Tuple.Extra     (second, first)
+#if MIN_VERSION_aeson(2,2,0)
+import           Data.Aeson           (FromJSON, Result (..), fromJSON)
+import           Data.Aeson.Parser    (json)
+#else
 import           Data.Aeson           (FromJSON, Result (..), fromJSON, json)
+#endif
```

Comment:
The split setup seems unnecessary. `Data.Aeson.Parser` exports the same function `json` as `Data.Aeson` in both versions 0.6.2.0 and 2.1.2.0. I didn't check intermediate versions, but it seems fairly unlikely this was different for some intermediate version, right?

So I think it can just always import `json` from `...Parser`.